source/common/sequencer_impl.cc

```diff
@@ -112,8 +112,14 @@ void SequencerImpl::run(bool from_periodic_timer) {
         const auto dur = time_source_.monotonicTime() - now;
         latency_statistic_->addValue(dur.count());
         targets_completed_++;
-        // Immediately schedule us to check again, as chances are we can get on with the next task.
-        spin_timer_->enableTimer(0ms);
+        // Callbacks may fire after stop(), as during teardown the dispatcher may run. (Concretely
+        // when BenchmarkClient::terminate() is called, and the connection pool shuts down). Hence
```

Comment:
Should we have an additional drain phase before termination, i.e. to empty out the dispatcher? It seems it would be easier to reason about this than arbitrary destructed state during the callback.

Reply:
I amended this to check for `this->running_` here instead? We only need to worry about things like this if `stop()` has been called, and that flag explicitly tracks that. Also attempted to improve the comment. WDYT?

Regarding a drain phase: imho ideally we'd model phases explicitly, and then define a (default) pre-termination phase via termination predicates that ensure we're bounded in time and exit with a failure code when we don't see the relevant gauges drop to 0 within that duration.